script.js

```diff
@@ -1,200 +1,70 @@
-// body
-const body = document.querySelector("body");
-//weight input
-const input = document.querySelector("#weightInput");
-// conversion output
-const output = document.querySelector("#output");
-// placeholder 
-const placeholder = input.placeholder;
-// select field
-const weightType = document.querySelector("#weightType")
-// output - grams 
-const gramsOutput = document.querySelector("#gramsOutput");
-// output - pounds
-const lbsOutput = document.querySelector("#lbsOutput");
-// output - ounces
-const ozOutput = document.querySelector("#ozOutput");
-// output - kilograms
-const kgOutput = document.querySelector("#kgOutput");
-// select field - kilograms
-const kilograms = document.querySelector("#kilograms");
-// select field - grams
-const grams = document.querySelector("#grams");
-// select field - lbs
-const pounds = document.querySelector("#lbs")
-// select field - oz
-const ounces = document.querySelector("#oz")
+function hideConversions() {
+    outputEl.style.visibility = 'hidden';
+}
 
-// second version of function for swapping placeholder text
-const changePlaceholder = () => {
-    switch (weightType.value) {
-        case `Kilograms`:
-            input.placeholder = "enter kilograms...";
-            break;
-        case `Grams`:
-            input.placeholder = `enter grams...`;
-            break;
-        case `Pounds`:
-            input.placeholder = "enter lbs...";
-            break;
-        case `Ounces`:
-            input.placeholder = "enter oz...";
-            break;
-        default:
-            input.placeholder = "enter kg...";
-    }
-};
+function showConversions() {
+    outputEl.style.visibility = 'visible';
+}
 
-// third version of conversion function - used switch for tidier code
-const calculateTwo = () => {
-      let weight = input.value;
-        if (input.value === "") {
-              output.style.visibility = 'hidden';
-          } else
-            switch (weightType.value) {
-                    case `Kilograms`:
-                        output.style.visibility = 'visible';
-                        gramsOutput.innerHTML = weight*1000; 
-                        lbsOutput.innerHTML = weight*2.20462;
-                        ozOutput.innerHTML = weight*35.274;
-                        kgOutput.innerHTML = weight*1;
+function changePlaceholder() {
+    inputEl.placeholder = `enter ${unitSelectorEl.value}`;
+}
+
+const weightOutput = () => {
+      let weight = inputEl.value;
+        if (inputEl.value === "") {
+            hideConversions();
+          } else {
+            showConversions();
+            switch (weightTypeEl.value) {
+                    case `kilograms`:
+                        gramsOutputEl.innerHTML = `${inputEl.value}kg = ${(weight*1000)}g`; 
+                        lbsOutputEl.innerHTML = `${inputEl.value}kg = ${(weight*2.20462).toFixed(3)} lb/s`; 
+                        ozOutputEl.innerHTML = `${inputEl.value}kg = ${(weight*35.274).toFixed(3)} oz`; 

```

Comment:
Okay this does give more context to someone reading your code. But this changes the UX of the feature. What if the designer doesn't want this and prefers the previous one?

`2.20462` and `35.274` are ratios right? What if we create a constant?
```
const kgRatios = {
  g: 1000,
  lb: 2.20462,
  oz: 35.274,
}
```

This way we can reuse these values everywhere. Imagine a codebase with `2.20462` everywhere? Won't it be a pain to keep track of it or make changes to it?

Reply:
`lbsOutputEl.innerHTML = kgRatios.lb * inputEl.value`